Sources/XCTest/TestFiltering.swift

```diff
@@ -42,12 +37,12 @@ internal struct TestFiltering {
 
     static func filterTests(entries: [XCTestCaseEntry], filter: TestFilter) -> [XCTestCaseEntry] {
         return entries
-            .map({ testCase, tests in
-                return (testCase, tests.filter({ filter(testCase, $0.0) }))
-            })
-            .filter({ testCase, tests in
+            .map { testCase, tests in
+                return (testCase, tests.filter { filter(testCase, $0.0) } )
+            }
+            .filter { testCase, tests in
                 return !tests.isEmpty
-            })
+            }
```

Comment:
So, I've searched some files in *swift/stdlib/public/core* to try to find occurrences of the string `"})"` (which would indicate the use of non-trailing syntax). I found some cases:

    return split(
      maxSplits: maxSplits,
      omittingEmptySubsequences: omittingEmptySubsequences,
      isSeparator: { $0 == separator })

---

    else if let offset = (e as? Int).map({ IntMax($0) }) ?? (e as? IntMax) { //... }

---

...

I'm not quite sure in terms of consistency, when it comes to use of trailing closures. 
But the general use of non-trailing syntax would indicate to me, that there are appropriate situations for its use. I'm therefore leaning towards the Rule Of Kevin.